### ISSUE TITLE
[BUG](testing): Highlight stale mods turned on after refreshing mod list

### ISSUE THREAD

**Issue by jiriks74** [aclist/dztui#170]
What it says on the tin.

When you go to `Options` => `List installed mods` and click `Highlight stale` the mods get red as expected.

But if you delete a mod the list gets refreshed and the button stays *clicked*/*turned on* while no mods are red. You need to click the button to turn the highlight off (even though nothing's highlighted) and then click it again to turn the highlight on.

**Comment by jiriks74** [comment]
Either of these is better. I, personally, think that the *in place* update would be better. I think that preserving the user choice until they make another is the way to go.

- If a user doesn't want to highlight stale - one click extra.
- if a user wants to keep highlight stale its a click every time the list gets reloaded.

**Comment by aclist** [comment]
Yes, I agree with you that the new list should not revert to a default state until the user explicitly chooses to unhighlight.

I'll be working on these fixes over the coming days. The Model is independent from the View, so it's trivial to make inline changes to the table without reloading, but the functionality for checking mods and returning to the table again touches on several parts of the code, especially considering edge cases like "no mods found" or "unable to delete mod". And the act of checking for stale mods is itself a time-expensive operation, so I need to test several scenarios before patching this.

**Comment by aclist** [comment]
The new beta updates the list in-place as you remove mods in the list, and the highlight status is preserved.

In conjunction with #169, this means that:

- You can highlight/unhighlight stale on the fly. This is calculated each time from scratch, since the user might have moved things around manually on their system.
- When you remove some or all mods from the list, these rows are removed in-place without refreshing the entire table.
- Over the course of fixing this, I noticed that statusbar tooltips for some table contexts (My servers, Recent servers) were not updating on-the-fly when rows were updated or removed, so I fixed this as well. This means you can remove mods and get an updated count in the statusbar.
- When an error occurs during deletion (mods could not be deleted, e.g., if the user forcibly removed the directory by hand after loading DZGUI), the table will reload from scratch>
- When exiting/re-entering the mods list context, buttons should revert to their default state and any non-applicable buttons should be removed.

This should give a more intuitive experience and I think it is working correctly, but by all means please try to break it and let me know.

**Comment by jiriks74** [comment]
Seems cool! I'll pull the changes and see how everything looks the next time I'll play.

**Comment by aclist** [comment]
This should be fully fixed and integrated with the latest version live on both stable and testing branches.

**Comment by jiriks74** [comment]
So far this seems to be fixed as well.
